Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp

```diff
@@ -0,0 +1,291 @@
+#include <AtomLyIntegration/CommonFeatures/Grid/GridComponentConstants.h>
+#include <AtomLyIntegration/CommonFeatures/Grid/GridComponentConfig.h>
+#include <AtomLyIntegration/CommonFeatures/Mesh/MeshComponentConstants.h>
+#include <AtomLyIntegration/CommonFeatures/Mesh/MeshComponentBus.h>
+#include <AtomLyIntegration/CommonFeatures/PostProcess/PostFxLayerComponentConstants.h>
+#include <AtomLyIntegration/CommonFeatures/PostProcess/ExposureControl/ExposureControlComponentConstants.h>
+#include <AtomLyIntegration/CommonFeatures/ImageBasedLights/ImageBasedLightComponentConstants.h>
+
+#include <EMStudio/AnimViewportRenderer.h>
+#include <EMotionFX/Source/EMotionFXManager.h>
+#include <EMotionFX/Source/ActorManager.h>
+#include <EMotionFX/CommandSystem/Source/CommandManager.h>
+#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
+
+
+namespace EMStudio
+{
+    static constexpr float DepthNear = 0.01f;
+
+    AnimViewportRenderer::AnimViewportRenderer(AZ::RPI::ViewportContextPtr viewportContext)
+        : m_windowContext(viewportContext->GetWindowContext())
+    {
+        // Create a new entity context
+        m_entityContext = AZStd::make_unique<AzFramework::EntityContext>();
+        m_entityContext->InitContext();
+
+        // Create the scene
+        auto sceneSystem = AzFramework::SceneSystemInterface::Get();
+        AZ_Assert(sceneSystem, "Unable to retrieve scene system.");
+        AZ::Outcome<AZStd::shared_ptr<AzFramework::Scene>, AZStd::string> createSceneOutcome = sceneSystem->CreateScene("AnimViewport");
+        AZ_Assert(createSceneOutcome, "%s", createSceneOutcome.GetError().data());
+        m_frameworkScene = createSceneOutcome.TakeValue();
+        m_frameworkScene->SetSubsystem<AzFramework::EntityContext::SceneStorageType>(m_entityContext.get());
+
+        // Create and register a scene with all available feature processors
+        // TODO: We don't need every procesors, only register the processor we are going to use.
+        AZ::RPI::SceneDescriptor sceneDesc;
+        m_scene = AZ::RPI::Scene::CreateScene(sceneDesc);
+        m_scene->EnableAllFeatureProcessors();
+
+        // Link our RPI::Scene to the AzFramework::Scene
+        m_frameworkScene->SetSubsystem(m_scene);
+
+        // Create a render pipeline from the specified asset for the window context and add the pipeline to the scene
+        AZStd::string defaultPipelineAssetPath = "passes/MainRenderPipeline.azasset";
+        AZ::Data::Asset<AZ::RPI::AnyAsset> pipelineAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
+            defaultPipelineAssetPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::Error);
+        m_renderPipeline = AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(pipelineAsset, *m_windowContext.get());
+        pipelineAsset.Release();
+        m_scene->AddRenderPipeline(m_renderPipeline);
+        m_renderPipeline->SetDefaultView(viewportContext->GetDefaultView());
```

Comment:
Might be worth checking if the asset loaded to fail and bring the plugin in some error state